bsd-user/qemu.h

```diff
@@ -553,6 +553,43 @@ static inline void *lock_user_string(abi_ulong guest_addr)
 #define unlock_user_struct(host_ptr, guest_addr, copy)          \
     unlock_user(host_ptr, guest_addr, (copy) ? sizeof(*host_ptr) : 0)
 
+#define CONCAT(a, b) CONCAT_(a,b)
+#define CONCAT_(a, b) a##b
+
+/* lock/unlock a byte vector of length len */
+#define WITH_LOCK_VEC(host_ptr, type, guest_addr, len)              \
+    for (abi_ulong brk = 0,                                         \
```

Comment:
When WITH_LOCK() is nested, then you declare abi_ulong brk in a nested fashion, leading to a warning when compiled with -Wshadow.

Reply:
We can do ```abi_ulong CONCAT(host_ptr, brk)```.

Reply:
I changed it, rebased and pushed.

Reply:
Excellent! Thank you! (It's been a holiday the last few days).
That was my thought and should fix the problem nicely.